src/pages/Portfolio/ImpactDashboard/ImpactDashboardPage.vue

```diff
@@ -77,9 +78,15 @@ export default {
 			allowedTeams: [],
 		};
 	},
+	mixins: [badgeGoalMixin],
 	apollo: {
-		async preFetch(config, client) {
-			return client.query({ query: portfolioQuery });
+		async preFetch(config, client, { route }) {
```

Comment:
Related to another comment, if we moved the saving logic to the client, this can also be moved to mounted.